### ISSUE TITLE
Support forking via Windows Subsystem for Linux (WSL) 2

### ISSUE THREAD

**Issue by mik3y64** [mschubert/clustermq#168]
First of all, thank you for such a great package. What a great package. Many of my colleagues are fully enjoying `clustermq`. Sadly, as a Windows user I can't take full advantage of the forking parallelism due to Windows being a non-Unix OS. With [Windows Subsystem for Linux (WSL) 2](https://devblogs.microsoft.com/commandline/announcing-wsl-2/) (not WSL 1, which is not a real kernel) to be released, which is real Linux kernel with Windows that will make full system call compatibility possible, is it possible to support forking and hence `multicore` parallelism in a native Unix-like environment for Windows? 

FYI I use R but don't have much knowledge about software or backend development so my issues might not make sense here. However, I dig up some stuffs about WSL and started to think about this possibility that if the users can [install WSL (like Ubuntu image)](https://docs.microsoft.com/en-us/windows/wsl/install-win10), clustermq can make its `multicore` work by pushing the computation to the WSL Linux-based environment and then send the result back? [Here is another resource I dig up about development in WSL](https://code.visualstudio.com/docs/remote/wsl). It depicts the architecture of how Windows and Linux subsystem can work together in the same machine, sharing the same OS, memory, CPU and even file system. 

It would be great if `clustermq` can make it work for Windows users and enable the use of forking parallelism for non-Unix system like Windows. It will also benefit the users of R [drake](https://cran.r-project.org/web/packages/drake/index.html) which uses clustermq as its main backend for HPC. Another approximation of forking for Windows is [parallelsugar package](https://github.com/nathanvan/parallelsugar) but it's not being developed anymore. Clustermq seems like the best bet for Windows users.

WSL2 is fully open source and they will work with developers interested in contributing to help get changes upstream. [Blog post writted by WSL2 team](https://devblogs.microsoft.com/commandline/shipping-a-linux-kernel-with-windows/) and their [github repo](https://github.com/microsoft/WSL2-Linux-Kernel). 

[WSL2 is now available for Windows insider](https://devblogs.microsoft.com/commandline/wsl-2-is-now-available-in-windows-insiders/)

[Deep dive - 1 hour talk by the team about WSL2](https://www.youtube.com/watch?v=lwhMThePdIo)

If you need anything please let me know I would like to help out testing or debugging.

If the support is impossible, do you have suggestion for workaround via WSL/WSL2 on local WIndows machine? Like installing WSL, and send the job to WSL Linux environment via a script manually or something then access the result via SSH. If possible, what's your thought on the performance difference compared to native `multicore` support on real Unix system Mac/Linux?

**Comment by mschubert** [comment]
Multicore/multiprocess is definitely something that I want to support on Windows.

Thank you for pointing me to the updates on WSL!

Related: #64 #84, #142, #166

**Comment by mik3y64** [comment]
Thank you mschubert! Looking forward to it!

I found this yesterday and it might be useful for you. Originally in Korean but I google translate it. [Using Linux R and Rstudio on Windows 10 via Windows Subsystem for Linux (WSL)](https://translate.google.com/translate?hl=en&sl=auto&tl=en&u=https%3A%2F%2Fecosang.github.io%2Fblog%2Fmiscellaneous%2Fwsl%2F)

_(In case you wonder if WSL works for clustermq, I also set it up WSL1 (not WSL2) using this guide, and install R and Rstudio, open a new rstudio session in the WSL1 environemnt, run clustermq with drake. Everything works but obviously it's not great experience because I have to run difference instances and so on.)_
